tests/UI/campaigns/functional/FO/09_productPage/03_writeReview.js

```diff
@@ -0,0 +1,356 @@
+
+      await expect(waitingApprovalReviewCount).to.be.greaterThan(0);
+    });
+
+    it('should check if review is in waiting for approval table', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkIfReviewInWaitingApproval', baseContext);
+
+      const tableData = await productcommentsModulePage.getReviewDataFromTable(page, 'waiting-approval');
+
+      await expect(tableData.title).to.contains(productReviewData.reviewTitle);
+      await expect(tableData.content).to.contains(productReviewData.reviewContent);
+      await expect(tableData.rating).to.contains(productReviewData.reviewRating);
+    });
+
+    it('should approve review', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'approveReview', baseContext);
+
+      await productcommentsModulePage.approveReview(page);
+      const newReviewCount = await productcommentsModulePage.getTableReviewCount(page, 'waiting-approval');
+
+      await expect(newReviewCount).to.equal(waitingApprovalReviewCount - 1);
+    });
+  });
+
+  describe('Check if review is now in "approved review" table', async () => {
+    it('should check if review is in "approved review" table', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkIfReviewIsApproved', baseContext);
+
+      const approvedReviewData = await productcommentsModulePage.getReviewDataFromTable(page, 'approved');
+
+      await expect(approvedReviewData.title).to.contains(productReviewData.reviewTitle);
+      await expect(approvedReviewData.content).to.contains(productReviewData.reviewContent);
+      await expect(approvedReviewData.rating).to.contains(productReviewData.reviewRating);
+    });
+
+    it('should logout from BO', async function () {
+      await loginCommon.logoutBO(this, page);
+    });
+  });
+
+  describe('Go to FO product detail page and check if review is displayed', async () => {
+    it('should open the shop page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goCheckIfReviewIsDisplayed', baseContext);
+
+      await foHomePage.goTo(page, global.FO.URL);
+
+      const result = await foHomePage.isHomePage(page);
+      await expect(result).to.be.true;
+    });
+
+    it('should go to the first product page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToProductPageToCheckReview', baseContext);
+
+      await foHomePage.goToProductPage(page, 1);
+
+      const pageTitle = await productPage.getPageTitle(page);
+      await expect(pageTitle.toUpperCase()).to.contains(ProductData.firstProductData.name);
+    });
+
+    it('should get, check and save the comment count', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'getCommentCountAfterApprove', baseContext);
+
+      const actualCommentCount = await productPage.getNumberOfComment(page);
```

Comment:
```suggestion
      const actualCommentCount = await productPage.getNumberOfComments(page);
```

Function name to change on the page